src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilter.java

```diff
@@ -1,21 +1,21 @@
 package org.broadinstitute.hellbender.engine.filters;
 
 import com.google.cloud.dataflow.sdk.transforms.SerializableFunction;
-import htsjdk.samtools.SAMRecord;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.MutableGATKRead;
 
-import java.io.Serializable;
 import java.util.Objects;
-import java.util.function.Function;
 import java.util.function.Predicate;
 
-@FunctionalInterface
+
 /**
- * Filters which operate on {@link SAMRecord} should implement this interface by overriding {@link #test(SAMRecord)}
+ * Filters which operate on {@link org.broadinstitute.hellbender.utils.read.GATKRead} should implement this interface by overriding {@link #test(org.broadinstitute.hellbender.utils.read.GATKRead)}
  *
  * ReadFilter extends Predicate and SerializableFunction.  It provides a default implementation of apply based on the
  * implmenting class's implementation of test().
  */
-public interface ReadFilter extends Predicate<SAMRecord>, SerializableFunction<SAMRecord, Boolean>{
+@FunctionalInterface
+public interface ReadFilter extends Predicate<GATKRead>, SerializableFunction<GATKRead, Boolean> {
```

Comment:
The type here may be an issue if I want to have a filter that works on mutable reads.